bindings/ergo-lib-wasm/src/transaction.rs

```diff
@@ -20,10 +20,154 @@ use std::convert::TryInto;
+            .map(CommitmentHintJson)
+            .map_err(to_js)
+            .unwrap();
+        Ok(CommitmentHint {
+            0: {
+                ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::CommitmentHint::try_from(
+                    commitment_hint_json.0
+                )
+                    .unwrap()
+            },
+        })
+    }
+}
+
+/// CommitmentHint JSON
+#[wasm_bindgen]
+pub struct CommitmentHintJson(ergo_lib::chain::json::hints::CommitmentHintJson);
+
+/// HintsBag
+#[wasm_bindgen]
+pub struct HintsBag(
+    pub(crate) ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::HintsBag,
+);
+
+#[wasm_bindgen]
+impl HintsBag {
+    /// Empty HintsBag
+    pub fn empty() -> HintsBag {
+        HintsBag {
+            0: ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::HintsBag::empty(),
+        }
+    }
+
+    /// Add commitment hint to the bag
+    pub fn add_commitment(&mut self, hint: CommitmentHint) {
+        self.0.add_hint(
+            ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::Hint::CommitmentHint(
+                hint.0,
+            ),
+        );
+    }
+
+    /// Length of HintsBag
+    pub fn len(&self) -> usize {
+        self.0.hints.len()
+    }
+
+    /// Get commitment
+    pub fn get(&self, index: usize) -> Result<CommitmentHint, JsValue> {
+        let commitment = self.0.commitments()[index].clone();
+        Ok(CommitmentHint(commitment))
+    }
+}
+
+impl From<ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::HintsBag> for HintsBag {
+    fn from(t: ergo_lib::ergotree_interpreter::sigma_protocol::prover::hint::HintsBag) -> Self {
+        HintsBag(t)
+    }
+}
+
+/// TransactionHintsBag
+#[wasm_bindgen]
+pub struct TransactionHintsBag(pub(crate) ergo_lib::wallet::multi_sig::TransactionHintsBag);
+
+#[wasm_bindgen]
+impl TransactionHintsBag {
+    /// Empty TransactionHintsBag
+    pub fn empty() -> TransactionHintsBag {
+        TransactionHintsBag {
+            0: ergo_lib::wallet::multi_sig::TransactionHintsBag::empty(),
+        }
+    }
+
+    /// Adding hints for input
+    pub fn add_hints_for_input(&mut self, index: usize, hints_bag: &HintsBag) {
+        self.0.add_hints_for_input(index, hints_bag.0.clone());
+    }
+
+    /// Outputting HintsBag corresponding for an input index
+    pub fn all_hints_for_input(&self, index: usize) -> HintsBag {
+        HintsBag::from(self.0.all_hints_for_input(index))
+    }
+}
+
+impl From<ergo_lib::wallet::multi_sig::TransactionHintsBag> for TransactionHintsBag {
+    fn from(t: ergo_lib::wallet::multi_sig::TransactionHintsBag) -> Self {
+        TransactionHintsBag(t)
+    }
+}
+
+/// Generate commitments for transaction
+#[wasm_bindgen]
+pub fn generate_commitments(
+    _state_context: &ErgoStateContext,
+    tx: &UnsignedTransaction,
+    boxes_to_spend: &ErgoBoxes,
+    data_boxes: &ErgoBoxes,
+    public_keys: Propositions,
```

Comment:
What do you think of moving `generate_commitments` from `Transaction` and `ReducedTransaction` to `Wallet`? This will eliminate the need of `public_keys` parameter since wallet already has them.

Reply:
Yes, good point! I moved them in the last commit.